osu.Game/Online/API/Requests/GetUserRequest.cs

```diff
@@ -2,18 +2,21 @@
 // See the LICENCE file in the repository root for full licence text.
 
 using osu.Game.Users;
+using osu.Game.Rulesets;
 
 namespace osu.Game.Online.API.Requests
 {
     public class GetUserRequest : APIRequest<User>
     {
         private readonly long? userId;
+        private readonly RulesetInfo ruleset;
 
-        public GetUserRequest(long? userId = null)
+        public GetUserRequest(long? userId = null, RulesetInfo ruleset = null)
         {
             this.userId = userId;
+            this.ruleset = ruleset;
         }
 
-        protected override string Target => userId.HasValue ? $@"users/{userId}" : @"me";
+        protected override string Target => userId.HasValue ? (ruleset != null ? $@"users/{userId}/{ruleset.ShortName}" : $@"users/{userId}") : @"me";
```

Comment:
What if we requested a null user id with non-null ruleset? isn't that what we should do when retrieving info of `me` with a specific ruleset?

Reply:
Can this not just be `users/{(userId ?? "me")}/{ruleset?.ShortName}`